### PR TITLE
ffmpeg: fix build for missing packet_internal.h

### PR DESCRIPTION
**Pull request by xuguangxin**
# Description
fix build for the latest ffmpeg code.

the build is broken by the following commit in ffmpeg

    build failed since following commit in ffmpeg

    commit 6e1903938befc0bce2f64e01770af2f65d625cfc
    Author: James Almer <jamrial@gmail.com>
    Date:   Tue Jun 2 18:38:33 2020 -0300

        avcodec/internal: move packet related functions to their own header

        Signed-off-by: James Almer <jamrial@gmail.com>
    commit 6e1903938befc0bce2f64e01770af2f65d625cfc
    Author: James Almer <jamrial@gmail.com>
    Date:   Tue Jun 2 18:38:33 2020 -0300

        avcodec/internal: move packet related functions to their own header

        Signed-off-by: James Almer <jamrial@gmail.com>

# Issue
NO

# Author(s)
 Xu Guangxin <guangxin.xu@intel.com>

# Performance impact

- [ ] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [x] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [ ] other
- [x] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

**Comment by xuguangxin** [comment]
@1480c1 could we turn off trailing space check for *.patch?
The space generated by git format-patch. It's no way to remove ..

 https://github.com/OpenVisualCloud/SVT-AV1/pull/1326/files#diff-280dcd3e39e9dfff92d2c1dc5e3a6387L634

**Comment by 1480c1** [comment]
> @1480c1 could we turn off trailing space check for *.patch?
> The space generated by git format-patch. It's no way to remove ..
> 
> https://github.com/OpenVisualCloud/SVT-AV1/pull/1326/files#diff-280dcd3e39e9dfff92d2c1dc5e3a6387L634

you should be able to remove that whitespace just fine, both git and patch accept it without that whitespace just fine

**Comment by xuguangxin** [comment]
@1480c1 thanks for replay, 
seems we can't. I use emacs to modify the patch. it will add following lines. 

```
diff --git a/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch b/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch
index 9115b6b12..005def9e8 100644
--- a/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch
+++ b/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch
@@ -82,7 +82,7 @@ new file mode 100644
 index 0000000000..0ae3dd325b
 --- /dev/null
 +++ b/libavcodec/libsvt_av1.c
-@@ -0,0 +1,549 @@
+@@ -0,1 +1,549 @@
 +/*
 +* Scalable Video Technology for AV1 encoder library plugin
 +*
@@ -632,6 +632,6 @@ index 0000000000..0ae3dd325b
 +    .caps_internal  = FF_CODEC_CAP_INIT_CLEANUP,
 +    .wrapper_name   = "libsvt_av1",
 +};
---
+--
 2.17.1

```

**Comment by 1480c1** [comment]
How about something simpler like
```bash
sed -i 's/[[:space:]]*$//' file.patch
```
?


also, you need to do something like `git clean -dffx` since just doing a hard reset will not remove dangling files

**Comment by xuguangxin** [comment]
seems it works. thanks, @1480c1 
but the FFmpeg build failed, do we use the latest FFmpeg? or we just pin down a version?

thanks

**Comment by 1480c1** [comment]
` git clone --branch release/4.2 --depth 1 https://github.com/FFmpeg/FFmpeg "$HOME/ffmpeg"` We currently have the github actions FFmpeg action pinned release/4.2, the appveyor one is the one still on master branch

File: ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1-with-svt-hevc.patch [hunk 1]
```diff
@@ -116,6 +116,7 @@ index 0000000000..5f1d8e033f
 +#include "libavutil/avassert.h"
 +
 +#include "internal.h"
++#include "packet_internal.h"
```
Comment (1480c1) [inline ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1-with-svt-hevc.patch:119]: please add the same check here as you did with the other patch, that way they are kept as closely synced as possible

**Comment by xuguangxin** [comment]
oh, sorry for miss this. fixed.

**Comment by xuguangxin** [comment]
also, it's it possible to move svt av1 a few lines later than svthevc? With this approach It will not conflict with svthevc anymore, we do not need two patches.

**Comment by 1480c1** [comment]
In my personal patches, I moved svt-av1 next to libaom and svt-vp9 next to libvpx so that way I can apply any of the patches in any order needed

**Comment by xuguangxin** [comment]
It's great, please help send a pr. It's very tedious to update two/three same patches every time.

thanks

**Comment by hassount** [comment]
is this PR good to go ?

**Comment by xuguangxin** [comment]
@1480c1, any more suggestion on this patch?

thanks

**Comment by xuguangxin** [comment]
thanks @1480c1 , @hassount we are good to go. 
thanks

**Comment by 1480c1** [comment]
@hassount 🏓 it would be better to review and get this in as soon as possible so that way people who want to build a newer version of FFmpeg to test with `libsvt_av1` can do so

**Pull request merged**